packages/blockchain-api/src/blockscout.ts

```diff
@@ -129,259 +176,149 @@ export class BlockscoutAPI extends RESTDataSource {
+
+      const block = new BigNumber(transferTx.blockNumber).toFixed()
+      const hash = transferTx.transactionHash
+      const timestamp = new Date(transferTx.timestamp).getTime()
+
+      switch (transfers.length) {
+        case 0: // Just a contract call
+          break
+
+        case 1: // It's a regular token transfer
+          {
+            const transfer = transfers[0]
+
+            const comment = transferTx.input ? formatCommentString(transferTx.input) : ''
+            const eventToAddress = transfer.toAddressHash.toLowerCase()
+            const eventFromAddress = transfer.fromAddressHash.toLowerCase()
+            const [type, address] = resolveTransferEventType(
+              userAddress,
+              eventToAddress,
+              eventFromAddress,
+              this.getAttestationAddress(),
+              this.getEscrowAddress()
+            )
+            events.push({
+              type,
               timestamp,
-            },
-            hash: txhash,
-          })
-        }
-
-        // Otherwise, it's a regular token transfer
-      } else {
-        const stableTokenTxs = transactions.filter(
-          (tx) => tx.contractAddress.toLowerCase() === this.stableTokenAddress && tx.value !== '0'
-        )
-
-        stableTokenTxs.sort((a, b) =>
-          new BigNumber(b.value).minus(new BigNumber(a.value)).toNumber()
-        )
-
-        if (stableTokenTxs.length < 1) {
-          return
-        }
-        const gasValue = new BigNumber(stableTokenTxs[0].gasUsed).multipliedBy(
-          new BigNumber(stableTokenTxs[0].gasPrice)
-        )
-        let event
-
-        switch (stableTokenTxs.length) {
-          // simple transfer
-          case 1:
-            event = stableTokenTxs[0]
-            break
-
-          // transfer with feeCurrency
-          // this figures out which 2 tx are the fee, and the other is the transfer
-          case 3:
-            const tx0Value = new BigNumber(stableTokenTxs[0].value)
-            const tx1Value = new BigNumber(stableTokenTxs[1].value)
-            const tx2Value = new BigNumber(stableTokenTxs[2].value)
-
-            if (tx0Value.plus(tx1Value).isEqualTo(gasValue)) {
-              event = stableTokenTxs[2]
-            } else if (tx0Value.plus(tx2Value).isEqualTo(gasValue)) {
-              event = stableTokenTxs[1]
+              block,
+              amount: {
+                // Signed amount relative to the account currency
+                value: new BigNumber(transfer.value)
+                  .multipliedBy(eventFromAddress === userAddress ? -1 : 1)
+                  .dividedBy(WEI_PER_GOLD)
+                  .toString(),
+                currencyCode: transfer.token,
+                timestamp,
+              },
+              address,
+              comment,
+              hash,
+            })
+          }
+          break
+
+        case 2: // Exchange events have two corresponding transfers (in and out)
```

Comment:
Could this large function be broken apart? I'm not sure having a big chunk of code in a switch case is best for readability

Reply:
Yep done 👍